lib/pbench/test/unit/server/test_schema.py

```diff
@@ -126,10 +155,88 @@ def test_invalid_required(self, test):
         ),
     )
     def test_invalid_optional(self, test):
+        """
+        Test parameter validation of an optional parameter
+        """
         x = Parameter("data", ParamType.STRING, required=False)
         json, expected = test
         assert x.invalid(json) is expected
 
+    @pytest.mark.parametrize(
+        "test",
+        (
+            ({"data": "yes"}, "Yes"),
+            ({"data": "NO"}, "No"),
+            ({"data": "MaYbE"}, "MAYBE"),
+            ({"data": "Maybe"}, "MAYBE"),
+        ),
+    )
+    def test_keyword_normalization(self, test):
+        """
+        Test parameter normalization for a keyword parameter.
+        """
+        x = Parameter("data", ParamType.KEYWORD, keywords=["Yes", "No", "MAYBE"])
```

Comment:
Shouldn't one of the keywords be in all lowercase?  (You have an all uppercase and a mixed-case, so it seems like `"Yes"` and `"No"` are redundant cases, here.)

Analogously, shouldn't you have parameterizations where both values _match_ in upper-, lower-, and mixed case, as well as various permutations of "mismatches"?